libs/orchestrator-ui-components/src/lib/components/SearchBar/searchQueryMapper.ts

```diff
@@ -0,0 +1,48 @@
+import { Value } from '@elastic/eui/src/components/search_bar/query/ast';
+import { QueryContainer } from '@elastic/eui/src/components/search_bar/query/ast_to_es_query_dsl';
+
+// Types to cast to since the QueryContainer defines the types of "match" and "simple_query_string" as "object"
+type MatchObject = { query: Value };
+type SimpleQueryStringObject = { query: string };
+
+export function mapEsQueryContainerToKeyValueTuple(
+    queryContainer: QueryContainer,
+): [string, string] | undefined {
+    if (queryContainer.match !== undefined) {
+        const firstKey: string = Object.keys(queryContainer.match)[0];
+
+        const firstValue = (
+            queryContainer.match[
+                firstKey as keyof typeof queryContainer.match
+            ] as MatchObject
+        ).query;
```

Comment:
<rant>

It feels a bit that something has gone terribly wrong in the industry if this is needed to get a simple match ;) Not even sure if this is inherently type safe or if it's just stating that it is typesafe by all this declaring/casting.

If I understand this correctly 'firstKey' is a string, and then becomes kinda typesafe by declaring that it is a key in this match. Which it was in the first place, since you got the key form the queryContainer match. But then again, I don't know enough of typescript to understand if this can be avoided.

I would have expected a oneliner, but got 10 lines instead.

</rant>

Reply:
This piece of code I am least proud of indeed. The problem is that when you use Object.keys you get an array of strings instead of an array of "keyof ..." therefore I used the typecast here. It gets even worse when I also need to cast the result to be able to access the query prop. Another solution I thought about is just adding a ts-ignore statement.